flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/tasks/CheckpointCoordinatorConfiguration.java

```diff
@@ -119,6 +123,14 @@ public int getTolerableCheckpointFailureNumber() {
 		return tolerableCheckpointFailureNumber;
 	}
 
+	public String getStateBackendName() {
+		return stateBackendName;
+	}
+
+	public void setStateBackendName(@Nonnull String newStateBackend) {
```

Comment:
Add a setter function here instead of passed in from constructor, because [`StreamGraph#getStateBackend()`](https://github.com/apache/flink/blob/f38106582e69765677ef8ebddece3a92641f6f22/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java#L895) may return null in `StreamingJobGraphGenerator#configureCheckpointing`